test/crowdsale/AllowanceCrowdsale.test.js

```diff
@@ -66,3 +68,13 @@ contract('AllowanceCrowdsale', function ([_, investor, wallet, purchaser, tokenW
     });
   });
 });
+
+contract('AllowanceCrowdsaleFail', function ([_, investor, wallet, purchaser, tokenWallet]) {
+  const rate = new BigNumber(1);
+  const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000';
+
+  it('should fail with token wallet different from token address', async function () {
+    this.token = await SimpleToken.new({ from: tokenWallet });
+    await AllowanceCrowdsaleFail.new(rate, wallet, this.token.address, ZERO_ADDRESS).should.be.rejectedWith(EVMRevert);
+  });
```

Comment:
Two things here:
- Can we move this scenario to a `describe` block within the `contract` block above?
- What about using `await assertRevert(...)` ?

Reply:
Absolutely, much better now, no need for `AllowanceCrowdsaleFail` either.